managedkafka/consumergroups/update_consumergroup.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package consumergroups
+
+// [START managedkafka_update_consumergroup]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"cloud.google.com/go/managedkafka/apiv1/managedkafkapb"
+	"google.golang.org/api/option"
+	"google.golang.org/protobuf/types/known/fieldmaskpb"
+
+	managedkafka "cloud.google.com/go/managedkafka/apiv1"
+)
+
+func updateConsumerGroup(w io.Writer, projectID, region, clusterID, consumerGroupID, topicPath string, partitionIndex, offset int, opts ...option.ClientOption) error {
```

Comment:
optional: I think it will be more useful if you demonstrate how to update multiple partitions at once, as getting a list of partition indices and their corresponding offsets.